src/main/java/org/opensearch/knn/quantization/quantizer/RandomGaussianRotation.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.quantizer;
+
+import lombok.experimental.UtilityClass;
+
+import java.util.Random;
+
+import static org.opensearch.knn.common.KNNConstants.QUANTIZATION_RANDOM_ROTATION_DEFAULT_SEED;
+
+@UtilityClass
+public class RandomGaussianRotation {
+
+    /**
+     * Generates a random rotation matrix using Gaussian distribution and orthogonalization.
```

Comment:
Maybe add comment about preservation of dot products and l2 distances when this is applied. Also add comment about seeding